src/Helpers/isNotNil.test.ts

```diff
@@ -0,0 +1,51 @@
+import { describe } from 'riteway'
+
+import { isNotNil } from './isNotNil'
+
+describe('isNotNil', async assert => {
+
+  const returnTrue = 'return true'
```

Comment:
Might just make this a general should, `return the correct boolean` so its not duplicate of the expected

Reply:
My 2 cents: I find it more readable to have the should right there, and to tell me exactly what it should be doing.

"Return the correct boolean" reads to me almost as "it should be working well and the return type should be a boolean"

Reply:
I used to do it that way, but I found that it created tests where the expected/should became out of sync over time. This is a very simple function so it's not likely but for more complicated functions where you might adjust/change/fix/addto the function and tests overtime they easily become out of sync and can cause confusion when reading the test results.

Reply:
I hear you, but wouldn't the added responsibility of keeping these should/given up to date correctly be part of the philosophy of RITEway?

I used to prefer the test code being the single source of truth and avoiding human-readable messages for the very same reason, but I'm not sure now, since the RITEway does feel like better documentation of purposes...

Reply:
R = Readable. I do believe that being explicit about the expected return in the should documents the test cases better. Remember, that reading the test output is also a way to see expected behavior:

```
ok 98 Given an empty string: should return true
ok 99 Given an empty array: should return true
ok 100 Given an empty object: should return true
ok 101 Given undefined: should return false
ok 102 Given null: should return false
ok 103 Given no value: should return false
```
I believe that more clearly documents the expected behavior than `return the correct boolean`

Reply:
Commenting on all of this here:

> I hear you, but wouldn't the added responsibility of keeping these should/given up to date correctly be part of the philosophy of RITEway?

> Readable. I do believe that being explicit about the expected return in the should documents the test cases better. Remember, that reading the test output is also a way to see expected behavior

> ok 98 Given an empty string: should return true
> ok 99 Given an empty array: should return true
> ok 100 Given an empty object: should return true
> ok 101 Given undefined: should return false
> ok 102 Given null: should return false
> ok 103 Given no value: should return false

How often do you actually read the output of all the passed tests? I don't ever, I mean they look nice and all when they pass but I ignore them. :)

Eric was the one that pointed it out to me to not duplicate the actual/expected in the given/should ( I don't believe he meant as a hard rule or anything, just a suggestion )

Its the failing tests we care about and that should answer the 5 questions, and those display the actual/expected in addition to given/should when they fails so I don't see a reason to duplicate the expected in the should. 

I don't think its right or wrong either way in my opinion, just one is easier to forget to update during maintenance. Feel free to duplicate the expected in the should if you would like, I won't complain I was just offering a suggestion :)

Reply:
its also possible that things like booleans and null, undefined are just too low level not complicated enough to not duplicate, so you both could very well be right and its better to name them in the given/should

Reply:
fwiw: on new libraries, I do look at the results of the tests to get a quick read of expected behavior. Once I'm familiar, then yeah, I just look at the failing tests.

I don't have an ultra strong opinion, but a mild preference. 😀

Reply:
> I don't have an ultra strong opinion, but a mild preference.

Same, I am willing to try out the more duplicated way if we want to do that as a team. :D

Reply:
Same here. Let's try that as a team and see how it goes.

Reply:
Sounds good, maybe il be proved wrong :D